gpfit/tests/t_examples.py

```diff
@@ -0,0 +1,82 @@
+import unittest
+import numpy as np
+import sys
+import os
+import importlib
+
+
+FILE_DIR = os.path.dirname(os.path.realpath(__file__))
+EXAMPLE_DIR = os.path.abspath(FILE_DIR+'../../../docs/source/examples')
+IMPORTED_EXAMPLES = {}
+
+
+class TestExamples(unittest.TestCase):
+
+    # To test a new example, add a function called `test_$EXAMPLENAME`, where
+    # $EXAMPLENAME is the name of your example in docs/source/examples without
+    # the file extension.
+    #
+    # This function should accept two arguments (e.g. 'self' and 'example').
+    # The imported example script will be passed to the second: anything that
+    # was a global variable (e.g, "sol") in the original script is available
+    # as an attribute (e.g., "example.sol")
+    #
+    # If you don't want to perform any checks on the example besides making
+    # sure it runs, just put "pass" as the function's body, e.g.:
+    #
+    #       def test_dummy_example(self, example):
+    #           pass
+    #
+    # But it's good practice to ensure the example's solution as well, e.g.:
+    #
+    #       def test_dummy_example(self, example):
+    #           self.assertAlmostEqual(example.sol["cost"], 3.121)
+    #
+
+    def test_t_ex6_1(self, example):
+        pass
+
```

Comment:
Oh, sorry, run it literally with `new_test(name, None)`

On Mon, Feb 1, 2016 at 3:19 PM, Michael Burton <notifications@github.com>
wrote:

> In gpfit/tests/t_examples.py
> <https://github.com/hoburg/gpfit/pull/26#discussion_r51474423>:
>
> > +    #
> > +    # If you don't want to perform any checks on the example besides making
> > +    # sure it runs, just put "pass" as the function's body, e.g.:
> > +    #
> > +    #       def test_dummy_example(self, example):
> > +    #           pass
> > +    #
> > +    # But it's good practice to ensure the example's solution as well, e.g.:
> > +    #
> > +    #       def test_dummy_example(self, example):
> > +    #           self.assertAlmostEqual(example.sol["cost"], 3.121)
> > +    #
> > +
> > +    def test_t_ex6_1(self, example):
> > +        pass
> > +
>
>             delattr(TestExamples, fn)  # delete the old old_test
>             new_name = "test_%s" % (name)
>             setattr(TestExamples, new_name, new_test(name))
>
> Error:
>
> ---------------------------------------------------------------------------
> TypeError                                 Traceback (most recent call last)
> /Users/mjburton11/Documents/SuperUROP/gpfit/gpfit/tests/t_examples.py in <module>()
>      47             delattr(TestExamples, fn)  # delete the old old_test
>      48             new_name = "test_%s" % (name)
> ---> 49             setattr(TestExamples, new_name, new_test(name))
>      50     TESTS.append(TestExamples)
>      51
>
> TypeError: new_test() takes exactly 2 arguments (1 given)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hoburg/gpfit/pull/26/files#r51474423>.
>